app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala

```diff
@@ -49,7 +51,8 @@ class EditSetRecordController @Inject() (
   editSetRecordEdit: editSetRecordEdit,
   editSetRecordEditDiscard: editSetRecordEditDiscard,
   editSetRecordEditSave: editSetRecordEditSave
-) extends MessagesAbstractController(messagesControllerComponents) with I18nSupport with Secured with FormSupport {
+)(implicit val ec: ExecutionContext)
+    extends MessagesAbstractController(messagesControllerComponents) with I18nSupport with Secured with FormSupport {
 
   private val logger: Logger = Logger(this.getClass)
   private lazy val legalStatuses: Seq[LegalStatus] = referenceDataService.getLegalStatuses
```

Comment:
On line 67, we should be using `findRecord`, as well.

Reply:
Done. We were not using `findRecord` in the save or discard actions either.

Reply:
Oh right.

Reply:
Confirmed.